### PR TITLE
Newmetric: NRMSE

### LINKED ISSUE #2303: Normalized Root Mean Squared Error (NRMSE) Calculation
🚀 Feature
Normalized Root Mean Squared Error (NRMSE) Calculation
Motivation
Calculating the Normalized Root Mean Squared Error (NRMSE) is a widely used metric in various fields, such as image processing, signal analysis, and machine learning. However, the absence of a built-in NRMSE function in our current toolkit poses challenges for users who require this metric for evaluating model performance and accuracy.

Pitch
I propose the addition of a native NRMSE calculation function to our toolkit, streamlining the process for users and enhancing the overall user experience. This feature would empower users to easily assess the performance of their models, fostering better-informed decision-making and refining their models for optimal results.

Alternatives
While users can currently implement custom code to calculate NRMSE, having a dedicated function within the toolkit would eliminate the need for additional development efforts. This would save time and resources for users who frequently utilize NRMSE in their work.

Additional context
The inclusion of NRMSE as a built-in feature aligns with our commitment to providing comprehensive tools for users across diverse domains. This enhancement would contribute to a more robust and user-friendly toolkit, catering to the needs of our community and ensuring a smoother workflow for those relying on NRMSE for their evaluations.

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #2303

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2442.org.readthedocs.build/en/2442/

### CODE REVIEW

**Comment by Borda** [comment]
@SkafteNicki, what is missing here to make it land? :chipmunk:

**Comment by SkafteNicki** [comment]
> @SkafteNicki, what is missing here to make it land? 🐿️

@Borda should be ready now :]

**Review by justusschock — APPROVED** [review]

File: src/torchmetrics/functional/__init__.py [hunk 1]
```diff
@@ -177,63 +177,65 @@
...
     "panoptic_quality",
+    "peak_signal_noise_ratio",
     "pearson_corrcoef",
     "pearsons_contingency_coefficient",
     "pearsons_contingency_coefficient_matrix",
     "permutation_invariant_training",
     "perplexity",
     "pit_permutate",
     "precision",
+    "precision_at_fixed_recall",
```
Comment (Borda) [inline src/torchmetrics/functional/__init__.py:195]: shall this be another PR?

File: src/torchmetrics/regression/nrmse.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+        >>> import torch
+        >>> from torchmetrics import NormalizedRootMeanSquaredError
+        >>> target = tensor([2.5, 5.0, 4.0, 8.0])
+        >>> preds = tensor([3.0, 5.0, 2.5, 7.0])
+        >>> nrmse = NormalizedRootMeanSquaredError(normalization="mean")
+        >>> nrmse(preds, target)
+        tensor(0.1919)
+        >>> nrmse = NormalizedRootMeanSquaredError(normalization="range")
+        >>> nrmse(preds, target)
```
Comment (Borda) [inline src/torchmetrics/regression/nrmse.py:110]:
```suggestion
        >>> nrmse(preds, target)
        tensor(0.1701)
```

**Comment by Borda** [comment]
@SkafteNicki it seems something wrong there is:
```
>>> import torch
>>> from torchmetrics import NormalizedRootMeanSquaredError
>>> target = torch.tensor([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]])
>>> preds = torch.tensor([[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]])
>>> nrmse = NormalizedRootMeanSquaredError(num_outputs=3)
>>> nrmse(preds, target)
Expected:
    tensor([1., 1., 1.])
Got:
    tensor([inf, inf, inf])

**Comment by Borda** [comment]
@SkafteNicki just a friendly ping :) seems almost done just revisit the examples...

**Comment by SkafteNicki** [comment]
> @SkafteNicki just a friendly ping :) seems almost done just revisit the examples...

yes, just need the final tests to pass :]

File: src/torchmetrics/functional/regression/nrmse.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        preds: Predicted tensor
+        target: Ground truth tensor
+        num_outputs: Number of outputs in multioutput setting
+        normalization: type of normalization to be applied. Choose from "mean", "range", "std"
+
+    """
+    sum_squared_error, num_obs = _mean_squared_error_update(preds, target, num_outputs)
+
+    target = target.view(-1) if num_outputs == 1 else target
+    if normalization == "mean":
```
Comment (Andrewwango) [inline src/torchmetrics/functional/regression/nrmse.py:38]: Please could we also have an L2 option which normalises by the sum of squares of target? This is a commonly used NMSE in image processing eg for MRI (eg see Eq. 5 in https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3947978/#sec-a.w.btitle )

Comment (Andrewwango) [reply src/torchmetrics/functional/regression/nrmse.py:38]: Hi @Borda , do you need any more input from me regarding this? Thanks

Comment (Borda) [reply src/torchmetrics/functional/regression/nrmse.py:38]: @SkafteNicki ^^

Comment (SkafteNicki) [reply src/torchmetrics/functional/regression/nrmse.py:38]: Support added for l2 in commit https://github.com/Lightning-AI/torchmetrics/pull/2442/commits/069354531ac911ab9e87e58e890d95552db58f27

File: src/torchmetrics/functional/__init__.py (hunk 1 above)
Comment (Borda) [reply src/torchmetrics/functional/__init__.py:195]: @SkafteNicki ^^
